resources/aws_ec2.go

```diff
@@ -65,11 +67,12 @@ var AwsRegions = []string{
 // AWS credentials must be present in ~/.aws - For detailed instructions see
 // http://docs.aws.amazon.com/cli/latest/userguide/cli-config-files.html
 type AwsEc2Res struct {
-	BaseRes `yaml:",inline"`
-	State   string `yaml:"state"` // state: running, stopped, terminated
-	Region  string `yaml:"region"`
-	Type    string `yaml:"type"` // the ec2 instance type, ie. t2.micro
-	ImageID string `yaml:"imageid"`
+	BaseRes  `yaml:",inline"`
+	State    string `yaml:"state"`    // state: running, stopped, terminated
+	Region   string `yaml:"region"`   // region must match an element of AwsRegions
+	Type     string `yaml:"type"`     // type of ec2 instance, ie. t2.micro
+	ImageID  string `yaml:"imageid"`  // imageid must be available on the chosen region
+	UserData string `yaml:"userdata"` // userdata is a path to a script run on first startup
```

Comment:
While AWS calls it this, feel free to rename this field to something better if you think that's suitable. Not saying it necessarily is in this case, but I'm open to ideas.

Reply:
I think LaunchScript would be appropriate.